x/evm/keeper/msg_server.go

```diff
@@ -1064,6 +1062,8 @@ func (s msgServer) CreateDeployToken(c context.Context, req *types.CreateDeployT
 		return nil, err
 	}
 
+	s.nexus.RegisterAsset(ctx, chain, nexus.NewAsset(req.Asset.Name, req.MinAmount))
```

Comment:
why are we now registering the asset before we confirm the token?

Reply:
if the confirm fails, the token is not in `confirmed` status. So we won't create any transfer on this token. The token goes back to `initialize` state, we need re-confirm it

Reply:
I see. But it feels weird to register an asset to the nexus module without the guarantee that it is actually useable in the chain. Can you remind me why we decided to go this way?